lib/rihanna/migration/upgrade.ex

```diff
@@ -0,0 +1,99 @@
+defmodule Rihanna.Migration.Upgrade do
+  @moduledoc """
+  A set of tools for upgrading an existing Rihanna jobs table.
+
+  Rihanna stores jobs in a table in your database. The default table name is
+  "rihanna_jobs". The name is configurable by either passing it as an argument
+  to the functions below or setting `:jobs_table_name` in Rihanna's config.
+
+  #### Using Ecto
+
+  The easiest way to upgrade the database is with Ecto.
+
+  Run `mix ecto.gen.migration upgrade_rihanna_jobs` and make your migration look
+  like this:
+
+  ```elixir
+  defmodule MyApp.UpgradeRihannaJobs do
+    use Rihanna.Migration.Upgrade
+  end
+  ```
+
+  Now you can run `mix ecto.migrate`.
+
+  #### Without Ecto
+
+  Ecto is not required to run Rihanna. If you want to upgrade the table yourself,
+  without Ecto, take a look at either `statements/0` or `sql/0`.
+
+  """
+
+  alias Rihanna.Migration.Upgrade
+
+  defmacro __using__(opts) do
+    table_name = Keyword.get(opts, :table_name, Rihanna.Config.jobs_table_name()) |> to_string
+
+    quote do
+      use Ecto.Migration
+
+      def up do
+        Enum.each(Upgrade.statements(unquote(table_name)), &execute/1)
+      end
+
+      def down do
+        execute("""
+        ALTER TABLE #{unquote(table_name)} DROP COLUMN due_at;
+        """)
+      end
+    end
+  end
+
+  @doc """
+  Returns a list of SQL statements that will upgrade the Rihanna jobs table if
+  executed sequentially.
+
+  By default it takes the name of the table from the application config.
+
+  You may optionally supply a table name as an argument if you want to override
+  this.
+
+  ## Examples
+
+      > Rihanna.Migration.Upgrade.statements
+      [...]
+
+      > Rihanna.Migration.Upgrade.statements("my_alternative_table_name")
+      [...]
+  """
+  @spec statements() :: list[String.t()]
+  @spec statements(String.t() | atom) :: list[String.t()]
+  def statements(table_name \\ Rihanna.Config.jobs_table_name())
+      when is_binary(table_name) or is_atom(table_name) do
+    [
+      # Postgres versions earlier than v9.6 do not suppport `IF EXISTS` predicates
+      # on alter table commands. For backwards compatibility we're using a try/catch
+      # approach to add the `due_at` column idempotently.
+      """
+      DO $$
+          BEGIN
```

Comment:
did you mean to double-nest the begin?

Reply:
Also, do we need to add this to the original migration as well?

Reply:
The double-nested `BEGIN` is what's used on the [How to add column if not exists on PostgreSQL?](https://stackoverflow.com/questions/12597465/how-to-add-column-if-not-exists-on-postgresql) StackOverflow answer.

Probably not needed on the original migration since we don't want user to run it more than once, but me may want to extend the upgrade migration in the future and have it run multiple times.